File: client/types/entry.ts
export type EntrySource = 'voice' | 'text';

export interface Entry {
  id: string;
  category: string;
  note: string;
  startTime: string;
  endTime: string;
  source: EntrySource;
  transcript?: string;
  audioUrl?: string;
  createdAt: string;
}

export interface CapturedEntryDraft {
  category: string;
  note: string;
  startTime: string;
  endTime: string;
}

File: ios-swift/Cronwatch/Models/Entry.swift
import Foundation

enum EntrySource: String, Codable { case voice, text }

struct Entry: Identifiable, Hashable, Codable {
    let id: String
    let captureId: String
    var category: String
    var note: String
    var startTime: Date
    var endTime: Date
    var source: EntrySource
    var transcript: String?
    var audioUrl: String?
    let createdAt: Date
}

struct CapturedEntryDraft: Codable, Equatable {
    var category: String
    var note: String
    var startTime: Date
    var endTime: Date
}

struct Capture: Identifiable, Equatable {
    let captureId: String
    let source: EntrySource
    var transcript: String?
    var audioUrl: String?
    let createdAt: Date
    var blocks: [Entry]

    var id: String { captureId }
}
